akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala

```diff
@@ -36,6 +37,469 @@ import scala.annotation.switch
     case _          ⇒ false
   }
 
+  private sealed trait ParserState {
+    def proceed(input: Byte, pp: JsonObjectParser): Unit
+  }
+
+  private object ParserState {
+
+    object UnknownState extends ParserState {
+      override def toString: String = "Unknown"
+
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit =
+        throw new FramingException(s"Invalid JSON encountered at position [${pp.pos}] of [${pp.buffer}] — unknown state")
+
+    }
+
+    object InitialState extends ParserState {
+      override def toString: String = "Initial"
+
+      @tailrec
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit = {
+        if (input == SquareBraceStart) {
```

Comment:
use pattern matching here since you do in all the other places

Reply:
good point ;)